connect/logger/logger.py

```diff
@@ -17,27 +17,29 @@
 logger = logging.getLogger()
 
 
-def function_log(custom_logger=None):
-    if not custom_logger:
-        custom_logger = logging.getLogger()
-        sformat = " %(levelname)-6s; %(asctime)s; %(name)-6s; %(module)s:%(funcName)s:line" \
-                  "-%(lineno)d: %(message)s"
-        for handler in custom_logger.handlers:
-            handler.setFormatter(logging.Formatter(sformat))
-
-    # noinspection PyUnusedLocal
-    def decorator(func, **kwargs):
-        # noinspection PyShadowingNames
-        @wraps(func)
-        def wrapper(self, *args, **kwargs):
-            custom_logger.info('Entering: %s', func.__name__)
-            custom_logger.debug('Function params: {} {}'.format(args, kwargs))
-            result = func(self, *args, **kwargs)
-            custom_logger.debug(
-                u'Function `{}.{}` return: {}'.format(
-                    self.__class__.__name__, func.__name__, result))
-            return result
-
-        return wrapper
-
-    return decorator
+class LoggerAdapter(logging.LoggerAdapter):
+    def __init__(self, logger_):
+        super(LoggerAdapter, self).__init__(logger_, {})
+        self.prefix = None
+
+    def process(self, msg, kwargs):
+        return (
+            '[%s] %s' % (self.prefix, msg) if self.prefix else msg,
+            kwargs
+        )
+
+    def setLevel(self, level):
+        self.logger.setLevel(level)
+
+
+def function_log(func):
```

Comment:
Before, function_log was accepting the custom_logger used by each Automation class. This has been simplified and is no longer neccessary.